portfolio/src/main/webapp/script.js

```diff
@@ -13,16 +13,46 @@
 // limitations under the License.
 
 /**
- * Adds a random greeting to the page.
+ * Gets a random greeting.
  */
-function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+function getRandomGreeting() {
+    const greetings =
+      ['Welcome!', '¡Bienvenido!', '歡迎!', 'Welkom!'];
 
   // Pick a random greeting.
   const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+  return greeting;
+}
+
+/** Allows for multiple calls to the typeWriter function, writes text on the webpage. */
+function writeText(inputText, container) {
+    var i = 0;
+    var txt = inputText;
+    var speed = 90;
 
-  // Add it to the page.
-  const greetingContainer = document.getElementById('greeting-container');
-  greetingContainer.innerText = greeting;
+    /** Types greeting on page. */
+    function typeWriter() {
```

Comment:
not too familiar with usage of nested functions. Is there a specific reason you are opting to use it here?

Reply:
I chose to nest the function because the function uses the variables (i, txt, speed) defined in the outer function.This is to ensure that the i value keeps incrementing and isn't reset with every call to the function. I tried passing in parameters to the function instead of nesting it, but the issue is that it would require a recursive structure that interferes with writing the whole word on the screen and being able to use the button multiple times. I can try making it recursive again, if necessary!

Reply:
I think what you have now is fine. 

You could consider an iterative approach and increase the delay for each char when calling setTimeout() which in my opinion is slightly more readable. 

```
function writeText(inputText, container) {
    var i = 0;
    var txt = inputText;
    var speed = 90;

    document.getElementById(container).innerHTML = "";
    
    for (i = 0; i < txt.length; i++) {
      appendInSetTimeoutWithDelay(container, txt.charAt(i), i*speed);
    }
}

function appendInSetTimeoutWithDelay(container, char, delay) {
  setTimeout(function(){
    document.getElementById(container).innerHTML += char;
  }, delay);
}
```

Reply:
Ok, I will definitely change it if I have time after implementing the basic features!